client/starwhale/api/_impl/data_store.py

```diff
@@ -54,22 +57,116 @@ def _check_move(src: str, dest: str) -> bool:
             return False
 
 
-class SwType:
-    def __init__(
-        self, name: str, pa_type: pa.DataType, nbits: int, default_value: Any
-    ) -> None:
+class SwType(ABC):
+    def __init__(self, name: str, pa_type: pa.DataType) -> None:
         self.name = name
         self.pa_type = pa_type
-        self.nbits = nbits
-        self.default_value = default_value
+
+    @abstractmethod
+    def merge(self, type: "SwType") -> "SwType":
+        ...
 
     def serialize(self, value: Any) -> Any:
         return value
 
     def deserialize(self, value: Any) -> Any:
         return value
 
-    def encode(self, value: Any) -> Optional[str]:
+    @staticmethod
+    def encode_schema(type: "SwType") -> Dict[str, Any]:
+        if isinstance(type, SwScalarType):
+            return {"type": str(type)}
+        if isinstance(type, SwListType):
+            return {
+                "type": "LIST",
+                "elementType": SwType.encode_schema(type.element_type),
+            }
+        if isinstance(type, SwObjectType):
+            ret = {
+                "type": "OBJECT",
+                "attributes": {
+                    k: SwType.encode_schema(v) for k, v in type.attrs.items()
+                },
+            }
+            if type.raw_type is Link:
+                ret["pythonType"] = "LINK"
+            else:
+                ret["pythonType"] = (
+                    type.raw_type.__module__ + "." + type.raw_type.__name__
+                )
+            return ret
+        raise RuntimeError(f"invalid type {type}")
+
+    @staticmethod
+    def decode_schema(schema: Dict[str, Any]) -> "SwType":
+        type_name = schema.get("type", None)
```

Comment:
`type_name = schema.get("type", None)` -> `type_name = schema.get("type")` ?

Reply:
just want to throw a customized error instead of the standard error